### ISSUE TITLE
upload.createSession fails hard if filename contains UTF-8 control characters

### ISSUE THREAD

**Issue by dluces** [OneDrive/onedrive-api-docs#692]
During a migration, I noticed one of the filenames had this control character for some reason `\uc281` (encoded as `%C2%81`). When creating the upload session, the endpoint failed hard and returned a 400 web page instead of an API error response (json).
While it's understandable that the API returns an error on such cases, it's probably better if it is handled more appropriately and a nicer API error is returned.

**Comment by ificator** [comment]
Thanks @dluces - I bet it's HTTP.sys rejecting the request before it even gets to our handler. We've had numerous issues with this kind of thing in the past and there's never a good answer I'm afraid :(

**Comment by dluces** [comment]
Yeah, that'd be more complicated. We're filtering out control characters now so it's not a problem for us anymore. But I thought you guys might wanna be aware of that 😃
